driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -664,16 +712,7 @@ public void onException(Connection connection, Exception exception, long latency
 
                 @Override
                 public boolean onTimeout(Connection connection, long latency, int retryCount) {
-                    QueryState queryState = queryStateRef.get();
```

Comment:
This piece seemed very similar to the root level `onTimeout()` - removing it did not affect any of our tests.